site/content/en/docs/manual/advanced/ai-tools.md

```diff
@@ -1,155 +1,309 @@
+
+## OpenCV
+
+The tool based on [Open CV](https://opencv.org/) Computer Vision library
+which is an open-source product that includes many CV algorithms.
+Some of these algorithms can be used to simplify the annotation process.
+
+To start working with OpenCV, you need to load the OpenCV library first.
+<br>To do this, on the [controls sidebar](/docs/manual/basics/controls-sidebar/),
+click **OpenCV**![OpenCV](/images/image201.jpg)
+<br>Library will be downloaded automatically.
+
+![](/images/image198.jpg)
+
+Once it is loaded, the tool's functionality will be available.
+
+### Intelligent scissors
+
+**Intelligent scissors** is a CV method of creating a polygon
+by placing points with the automatic drawing of a line between them.
+The distance between the adjacent points is limited by the threshold of action,
+displayed as a red square that is tied to the cursor.
+
+To use **Intelligent scissors**, do the following:
+
+1. In the **OpenCV** menu, go to the **Drawing** tab.
+2. Select the label, and then click on the **Intelligent scissors** button.
+
+![](/images/image199.jpg)
+
+3. Add the first point on the boundary of the allocated object. <br> You will see a line repeating the outline of the object.
+4. Add the second point, so that the previous point is within the restrictive threshold.
+   <br>After that a line repeating the object boundary will be automatically created between the points.
+   ![](/images/image200_detrac.jpg)
+5. To finish placing points, on the top menu**Done** (or **N** on the keyboard).
 
-Fast online Object Tracking and Segmentation. Tracker is able to track different objects in one server request.
-Trackable object will be tracked automatically if the previous frame was
-a latest keyframe for the object. Have tracker indication on canvas. `SiamMask` tracker supported CUDA.
+As a result, a polygon will be created
+For more information, see [Annotation with polygons](/docs/manual/advanced/annotation-with-polygons/).
 
-> If you plan to track simple non-overlapping objects consider using fast client-side [TrackerMIL from OpenCV](/docs/manual/advanced/opencv-tools/#trackermil).
+To increase or lower the action threshold, hold **Ctrl** and scroll the mouse wheel.
+Increasing the action threshold will affect the performance.
+During the drawing process, you can remove the last point by clicking on it with the left mouse button.
+
+When tracking:
+
+- You can create a boundary manually (like when
+  [creating a polygon](/docs/manual/advanced/annotation-with-polygons/manual-drawing/))
+  by temporarily disabling the automatic line creation. To do that, click `Ctrl`.
+
+- You can change the number of points in the polygon with slider:
+
+  ![](/images/image224.jpg)
+
+- To change polygon opacity, on the [Objects sidebar](/docs/manual/basics/objects-sidebar/#appearance)
+  use the **Selected opacity** slider.
+
+### Histogram Equalization
```

Comment:
I will suggest creating a separate page and description of how we can adjust our original frame. Histogram Equalization is about that.

<img width="797" alt="image" src="https://user-images.githubusercontent.com/40690625/206844772-0f1a3a7b-c437-4ec5-8888-84facd9fdd99.png">

Reply:
On the same page we can description the image grid, brightness, contract, and saturation.

Reply:
I agree, but suggest making different task/request for this one